VirtoCommerce.CartModule.Data/Repositories/CartRepositoryImpl.cs

```diff
@@ -183,28 +183,41 @@ protected override void OnModelCreating(DbModelBuilder modelBuilder)
 
         public virtual ShoppingCartEntity[] GetShoppingCartsByIds(string[] ids, string responseGroup = null)
         {
-            var carts = ShoppingCarts.Where(x => ids.Contains(x.Id)).ToArray();
+            ShoppingCartEntity[] result;
 
-            var cartTaxDetails = TaxDetails.Where(x => ids.Contains(x.ShoppingCartId)).ToArray();
-            var cartDiscounts = Discounts.Where(x => ids.Contains(x.ShoppingCartId)).ToArray();
-            var cartAddresses = Addresses.Where(x => ids.Contains(x.ShoppingCartId)).ToArray();
-            var cartCoupons = Coupons.Where(x => ids.Contains(x.ShoppingCartId)).ToArray();
-
-            var paymentIds = Payments.Include(x => x.Addresses)
-                                    .Where(x => ids.Contains(x.ShoppingCartId)).ToArray().Select(x => x.Id).ToArray();
-            var paymentTaxDetails = TaxDetails.Where(x => paymentIds.Contains(x.PaymentId)).ToArray();
-            var paymentDiscounts = Discounts.Where(x => paymentIds.Contains(x.PaymentId)).ToArray();
-
-            var lineItemIds = LineItems.Where(x => ids.Contains(x.ShoppingCartId)).ToArray().Select(x => x.Id).ToArray();
-            var lineItemsTaxDetails = TaxDetails.Where(x => lineItemIds.Contains(x.LineItemId)).ToArray();
-            var lineItemsDiscounts = Discounts.Where(x => lineItemIds.Contains(x.LineItemId)).ToArray();
+            if (ids.IsNullOrEmpty())
+            {
+                result = new ShoppingCartEntity[0];
+            }
+            else
+            {
+                result = ShoppingCarts.Where(x => ids.Contains(x.Id)).ToArray();
 
-            var shipmentIds = Shipments.Include(x => x.Items).Where(x => ids.Contains(x.ShoppingCartId)).ToArray().Select(x => x.Id).ToArray();
-            var shipmentTaxDetails = TaxDetails.Where(x => shipmentIds.Contains(x.ShipmentId)).ToArray();
-            var shipmentDiscounts = Discounts.Where(x => shipmentIds.Contains(x.ShipmentId)).ToArray();
-            var shipmentAddresses = Addresses.Where(x => shipmentIds.Contains(x.ShipmentId)).ToArray();
+                if (result.Any())
+                {
+                    ids = result.Select(x => x.Id).ToArray();
+                    var cartTaxDetails = TaxDetails.Where(x => ids.Contains(x.ShoppingCartId)).ToArray();
+                    var cartDiscounts = Discounts.Where(x => ids.Contains(x.ShoppingCartId)).ToArray();
+                    var cartAddresses = Addresses.Where(x => ids.Contains(x.ShoppingCartId)).ToArray();
+                    var cartCoupons = Coupons.Where(x => ids.Contains(x.ShoppingCartId)).ToArray();
+
+                    var paymentIds = Payments.Include(x => x.Addresses)
```

Comment:
Need to add the same IsNullOrEmpty check for paymentIds

Reply:
Fixed